### PR TITLE
NPC trade ability fixes + other missing npc dialogs

### PR DESCRIPTION
**Pull request by nickolasdeluca**
Implementing Melfar's trade shop and fixing Shanar not buying Orcish axes.

### CODE REVIEW

**Comment by oalbsilva** [comment]
fix #1258
fix #1263

**Review by YohanAugusto — APPROVED** [review]

File: data/npc/scripts/hofech.lua [hunk 1]
```diff
@@ -15,4 +15,92 @@ function onThink()
     npcHandler:onThink()
 end
 
+local shopModule = ShopModule:new()
+npcHandler:addModule(shopModule)
+
+shopModule:addBuyableItem({'amphora'}, 2023, 4, 'amphora')
```
Comment (dudantas) [inline data/npc/scripts/hofech.lua]: The second name is not necessary, instead you can put the item quantity, example: "1"

Comment (dudantas) [reply data/npc/scripts/hofech.lua]: Look here: https://github.com/opentibiabr/otservbr-global/blob/10ee1bc11df548302296015cbbc4f2b3650aa2b8/data/npc/scripts/richard.lua#L33

Comment (nickolasdeluca) [reply data/npc/scripts/hofech.lua]: k, that's easy to fix, I'll remove them and commit.

**Review by dudantas — APPROVED** [review]

**Comment by dudantas** [comment]
Thanks for your contribution!

**Pull request merged**